compiler/src/dotty/tools/dotc/core/TypeEval.scala

```diff
@@ -94,6 +97,22 @@ object TypeEval:
             throw TypeError(em"${e.getMessage.nn}")
         ConstantType(Constant(result))
 
+      def fieldsOf: Option[Type] =
+        expectArgsNum(1)
+        val arg = tp.args.head
+        val cls = arg.classSymbol
+        if cls.is(CaseClass) then
+          val fields = cls.caseAccessors
+          val fieldLabels = fields.map: field =>
+            ConstantType(Constant(field.name.toString))
+          val fieldTypes = fields.map(arg.memberInfo)
+          Some:
+            defn.NamedTupleTypeRef.appliedTo:
+              nestedPairs(fieldLabels) :: nestedPairs(fieldTypes) :: Nil
```

Comment:
```suggestion
            defn.NamedTuple(nestedPairs(fieldLabels), nestedPairs(fieldTypes))
```

Reply:
The proposed replacement does not compiler. There is no `NamedTuple` method in `Definitions`.

Reply:
there's a `NamedTuple` object with apply method that's being used in some other places